bindings/rust/src/bindings/mod.rs

```diff
@@ -194,6 +198,64 @@ impl Drop for KZGSettings {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct Sample(Vec<Bytes32>);
+
+impl Sample {
+    pub fn as_slice(&self) -> &[Bytes32] {
+        self.0.as_slice()
+    }
+
+    pub fn from_field_elements(
+        elements: &[Bytes32],
+        kzg_settings: &KZGSettings,
+    ) -> Result<Self, Error> {
+        if elements.len() != kzg_settings.sample_size as usize {
+            return Err(Error::InvalidFieldElementsLength);
+        }
+        Ok(Self(Vec::from(elements)))
+    }
+}
+
+pub struct SampleMatrix<const ROWS: usize, const COLS: usize> {
+    matrix: Vec<Vec<Sample>>,
+}
+
+impl<const ROWS: usize, const COLS: usize> SampleMatrix<ROWS, COLS> {
+    pub fn entry(&self, row: usize, col: usize) -> Option<&Sample> {
+        if row >= ROWS || col >= COLS {
+            return None;
+        }
+
+        Some(&self.matrix[row][col])
+    }
```

Comment:
So there's nothing stopping someone from creating an invalid sample matrix and calling this function.

```rust
let m = SampleMatrix::<20, 20> {
    matrix: vec![vec![sample; 5]; 5],
};
```

If someone were to call `m.entry(10, 10)` there would be a problem.